File: app/src/main/java/com/kakacat/minitool/wifipasswordview/MyAdapter.java
package com.kakacat.minitool.wifipasswordview;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.kakacat.minitool.R;
import com.kakacat.minitool.common.myinterface.RecycleViewItemOnLongClickListener;
import com.kakacat.minitool.common.ui.view.CircleProgressView;

import java.util.List;

public class MyAdapter extends RecyclerView.Adapter<MyAdapter.ViewHolder> {

    private LayoutInflater inflater;
    private List<Wifi> wifiList;
    private RecycleViewItemOnLongClickListener longClickListener;

    public MyAdapter(List<Wifi> wifiList) {
        this.wifiList = wifiList;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        if(inflater == null) {
            inflater = LayoutInflater.from(parent.getContext());
        }
        View view = inflater.inflate(R.layout.wifi_item_layout,parent,false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        Wifi wifi = wifiList.get(position);
        holder.circleProgressView.setText(wifi.getWifiImage());
        holder.tvWifiName.setText(wifi.getWifiName());
        holder.tvWifiPwd.setText(wifi.getWifiPwd());
        if(longClickListener != null){
            holder.itemView.setOnLongClickListener(v -> {
                longClickListener.onLongClick(v,position);
                return false;
            });
        }
    }

    @Override
    public int getItemCount() {
        return wifiList.size();
    }

    public void setLongClickListener(RecycleViewItemOnLongClickListener longClickListener){
        this.longClickListener = longClickListener;
    }


    static class ViewHolder extends RecyclerView.ViewHolder{

        private CircleProgressView circleProgressView;
        private TextView tvWifiName;
        private TextView tvWifiPwd;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            circleProgressView = itemView.findViewById(R.id.circle_progress_view);
            tvWifiName = itemView.findViewById(R.id.tv_wifi_name);
            tvWifiPwd = itemView.findViewById(R.id.tv_wifi_pwd);
        }
    }

}

File: app/src/main/java/com/kakacat/minitool/garbageclassify/MyAdapter.kt
package com.kakacat.minitool.garbageclassify

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.kakacat.minitool.R
import com.kakacat.minitool.common.ui.RecycleViewListener.OnItemClick
import com.kakacat.minitool.garbageclassify.model.Garbage
import com.kakacat.minitool.garbageclassify.model.TypeMap.getIcon
import com.kakacat.minitool.garbageclassify.model.TypeMap.getTypeName

class MyAdapter(private val garbageList: List<Garbage>) : RecyclerView.Adapter<MyAdapter.ViewHolder>() {

    private var listener: OnItemClick? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.garbage_layout, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val garbage = garbageList[position]
        val type = garbage.type
        holder.tvName.text = garbage.name
        holder.tvType.text = getTypeName(type)
        holder.ivIcon.setBackgroundResource(getIcon(type))
        if (listener != null) {
            holder.itemView.setOnClickListener { listener!!.onClick(holder.itemView, position) }
        }
    }

    override fun getItemCount(): Int {
        return garbageList.size
    }

    fun setOnClickListener(listener: OnItemClick?) {
        this.listener = listener
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val ivIcon: ImageView = itemView.findViewById(R.id.iv_icon)
        val tvName: TextView = itemView.findViewById(R.id.tv_name)
        val tvType: TextView = itemView.findViewById(R.id.tv_type)
    }
}